src/main/java/core/basesyntax/CalculationFruits.java

```diff
@@ -0,0 +1,35 @@
+package core.basesyntax;
+
+import core.basesyntax.service.CalculationFile;
+import core.basesyntax.service.FruitsDto;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class CalculationFruits implements CalculationFile {
+    Map<String, Integer> calculation = new HashMap<>();
+
+    @Override
+    public Map<String, Integer> getCalculation(List<FruitsDto> data) {
+        for (FruitsDto value : data) {
+            if (calculation.containsKey(value.getFruit())) {
+                if (value.getType().contains("b")) {
+                    if (calculation.get(value.getFruit()) < value.getQuantity()) {
+                        calculation.put("File has incorrect data! They not include to result!\n"
+                                + value.getType() + " " + value.getFruit()
+                                + " " + value.getDate(), value.getQuantity());
+                    } else {
+                        calculation.put(value.getFruit(),
+                                calculation.get(value.getFruit()) - value.getQuantity());
+                    }
+                } else {
+                    calculation.put(value.getFruit(),
+                            calculation.get(value.getFruit()) + value.getQuantity());
+                }
```

Comment:
I think it's possible to avoid the repetition of the code here, e.g. it could be put into a separate method if there's no way to refactor the if-else structure here